src/client/logging/_global.ts

```diff
@@ -70,7 +71,15 @@ function initialize() {
     }
 }
 
-// Register the output channel transport the logger will log into
+// Set the logging level the extension logs at.
+export function setLoggingLevel(level: LogLevel) {
+    const levelName = resolveLevelName(level, winston.config.npm.levels);
+    if (levelName) {
+        globalLogger.level = levelName;
+    }
```

Comment:
If the level is "Off" then we need to disable logging:

```suggestion
    if (level === LogLevel.Off) {
        // Disable logging.
        globalLogger.clear();
    } else {
        const levelName = resolveLevelName(level, winston.config.npm.levels);
        if (levelName) {
            globalLogger.level = levelName;
        }
    }
```

It may make sense to have the extension setting only apply to the output channel transport (once we add it).  However, we can address that separately.